src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -83,4 +83,16 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+   default String getPrintableString(Printable... printables) {
+	   final StringBuilder b = new StringBuilder();
+	   for(Printable p : printables) {
+		   b.append(p);
```

Comment:
you need to use `p.getPrintableString()` here. Don't think this will work since Printable does not explicitly indicate that it implements `toString`